utils/frame/try-runtime/cli/tests/on_runtime_upgrade.rs

```diff
@@ -0,0 +1,72 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg(unix)]
+
+#[cfg(feature = "try-runtime")]
+mod tests {
+	use assert_cmd::cargo::cargo_bin;
+	use regex::Regex;
+	use std::{process, time::Duration, str::from_utf8};
+	use substrate_cli_test_utils as common;
+	use tokio::process::Command;
+
+	#[tokio::test]
+	async fn on_runtime_upgrade_works() {
+		// Build substrate so binaries used in the test use the latest code.
+		common::build_substrate(&["--features=try-runtime"]);
```

Comment:
Why is this needed?  
The check above should already ensure that, or `feature = "try-runtime"`?  
I dont think we should need to build withing a test.

Reply:
@ggwpez we build within the test because we spin up an actual substrate binary in the test

If we don't build prior to running, the binary used may not match the code on disk

Reply:
Removed the `#[cfg(feature = "try-runtime")]` feature flag as I do not see any difference, tests still work without it.

Reply:
Weird, since the [node CLI tests](https://github.com/paritytech/substrate/blob/a1d3d5004379976b13deab86b975c04ad78bcd72/bin/node/cli/tests/benchmark_block_works.rs#L33) do not need to re-build.  
Maybe the tests are in the wrong directory?